sm/plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -73,27 +99,33 @@ fn run(operation: PluginOp) -> Result<ExitStatus, InternalError> {
             status
         }
 
-        PluginOp::Install { module, version } => {
-            if let Some(version) = version {
-                run_cmd(
-                    "apt-get",
-                    &format!("install --quiet --yes {}={}", module, version),
-                )?
-            } else {
-                run_cmd("apt-get", &format!("install --quiet --yes {}", module))?
+        PluginOp::Install { module, version } => install(module, version)?,
+
+        PluginOp::Remove { module, version } => remove(module, version)?,
+
+        PluginOp::UpdateList => {
+            let mut updates: Vec<SoftwareModuleUpdate> = Vec::new();
+            let mut status: ExitStatus = ExitStatus::from_raw(0);
+            let mut rdr = csv::ReaderBuilder::new()
+                .has_headers(false)
+                .delimiter(b' ')
+                .from_reader(io::stdin());
+            for result in rdr.deserialize() {
+                updates.push(result?);
             }
-        }
 
-        PluginOp::Remove { module, version } => {
-            if let Some(version) = version {
-                // check the version mentioned present or not
-                run_cmd(
-                    "apt-get",
-                    &format!("remove --quiet --yes {}={}", module, version),
-                )?
-            } else {
-                run_cmd("apt-get", &format!("remove --quiet --yes {}", module))?
+            print!("{:?}", updates);
+            for update_module in updates {
+                let res = match update_module.action {
+                    UpdateAction::Install => install(update_module.name, update_module.version),
+                    UpdateAction::Remove => remove(update_module.name, update_module.version),
+                };
+                status = res?;
+                if !status.success() {
+                    break;
+                }
```

Comment:
```suggestion
                let status = match update_module.action {
                    UpdateAction::Install => install(update_module.name, update_module.version)?,
                    UpdateAction::Remove => remove(update_module.name, update_module.version)?,
                };

                if !status.success() {
                    return Ok(status);
                }
            }
            ExitStatus::from_raw(0)
```

I think this makes the intentions a little bit cleaner.